rust/datafusion/tests/sql.rs

```diff
@@ -2528,7 +2528,7 @@ async fn in_list_array() -> Result<()> {
             ,c1 IN ('x', 'y') AS utf8_in_false
             ,c1 NOT IN ('x', 'y') AS utf8_not_in_true
             ,c1 NOT IN ('a', 'c') AS utf8_not_in_false
-            ,CAST(CAST(c1 AS int) AS varchar) IN ('a', 'c') AS utf8_in_null
```

Comment:
I don't really understand what the original test was testing 🤔  but the change looks better

Reply:
This was a hack to create a NULL of UTF8 type (given the previous CAST semantics the failed cast would be NULL)